### ISSUE TITLE
HTODemux error "Cells with zero counts exist as a cluster"

### ISSUE THREAD

**Issue by simonmfr** [satijalab/seurat#5640]
I keep getting the following error from Seurat::HTODemux(): `Cells with zero counts exist as a cluster.`

As previously discussed in #3557 #4435 #2549, I removed cells with 0 HTO counts and HTOs with few (up to 800) counts, but keep getting the error.

My question: If I don't normalize the HTO assay before running HTODemux, the function seems to work. Would these results be valid?

Appreciate any reply!

**Comment by mhkowalski** [comment]
Hi, 
 
You could try changing ```nsamples = 1000``` within the ```HTODemux()``` command (assuming you have more than 1000 cells). You could also simply add 1 to the HTO count matrix, such as 

```
m <- GetAssayData(obj, assay="HTO") 
m <- m +1 
new_assay <- CreateAssayObject(counts=m)
obj[['HTOnew']] <- new_assay
```
You should then run ```HTODemux()```  using the new assay after normalizing. 

Will you let me know if either of these strategies work for you? 

Thanks,
Madeline

**Comment by simonmfr** [comment]
Adding +1 to each HTO count solves the problem, many thanks. I assume the demultiplexing outcome is not affected since 1 is added to all values?

Setting `nsamples = 1000` did not help despite having >1000 cells, just fyi.

Much appreciated!
Simon

**Comment by segalmicrobiomelab** [comment]
I was talking to Igor and having the same problem - one issue was leftover HTO/CMO tags that were included, but not used, although some cells/pathways were assigned, after filtering these out I was able to run HTOdemux(). The part where it was fixed was after loading the data after read10x() portion and before the Assay() and HTOdemux() functions. I filtered out all of the non used HTO/CMO from the read10xobjects$'Antibody Marker' matrix.
